postgres-openssl/src/lib.rs

```diff
@@ -134,13 +134,15 @@ where
     S: AsyncRead + AsyncWrite + Unpin + Debug + 'static + Sync + Send,
 {
     type Stream = TlsStream<S>;
-    type Error = HandshakeError<S>;
+    type Error = Error;
     #[allow(clippy::type_complexity)]
-    type Future = Pin<Box<dyn Future<Output = Result<TlsStream<S>, HandshakeError<S>>> + Send>>;
+    type Future = Pin<Box<dyn Future<Output = Result<TlsStream<S>, Error>> + Send>>;
 
     fn connect(self, stream: S) -> Self::Future {
         let future = async move {
-            let stream = tokio_openssl::connect(self.ssl, &self.domain, stream).await?;
+            let ssl = self.ssl.into_ssl(&self.domain)?;
+            let mut stream = Box::pin(SslStream::new(ssl, stream)?);
```

Comment:
Since `S` is `Unpin`, `SslStream<S>` is as well so there's no need to use `Box::pin` here. You should be able to just do `Pin::new(&mut stream).connect().await;`

Reply:
Ah yeah, thanks, that's much nicer. I missed the `Unpin` bound on `S`.